gips/0066-graph-horizon.md

```diff
@@ -0,0 +1,348 @@
+    
+- **Delegation Cuts**
+    
+    Horizon allows service providers to define delegation cuts for any payment type defined in the payments protocol. This ensures delegators can be rewarded differently depending on the type of work they are supporting with their delegation. For more details see the payments protocol section.
+    
+
+Note that additional functionality can be built on top of these core mechanisms. We’ve seen that happen already in the current iteration of The Graph, ideas like liquid delegation, delegation aggregators or pools, etc, could be implemented by third parties or core dev teams in a later stage.
+
+### Slashing
+
+In order for provisioned stake to be considered useful economic security it needs to be slashable. The fact that stake is at risk disincentivizes service providers from ill intentioned behavior, furthermore, the slashing mechanism uses part of the slashed stake as a reward for reporting faults. It then follows that *all stake* that is being used as economic security for a service should be slashable, not just a portion of it. Graph Horizon leaves it to the data service to decide why and when a service provider is slashed and what the slashed amount is. As far as the staking protocol is concerned a data service could slash the entire provisioned stake (both from the service provider and delegators) in a single slashing event. Graph Horizon’s guarantees that part of the slashed amount will be burnt and another part sent back to the data service as the reward. It's important to be aware that a compromised or malicious data service could potentially misuse this mechanism to access funds. Community standards and best practices such as those described by the Data Service framework should help curate and identify those that are safe to use.
+
+As described before, the slashing risk applies both to stake provisioned by the service provider and delegated stake. To mitigate the risk of service providers misappropriating delegator funds by using a malicious data service via slashing, we introduce a few protections for delegators. First, the staking contract ensures that all ****service provider stake will be slashed before the delegator’s stake is touched. Since a portion of the slashed amount is burned it greatly reduces the economic viability of an attack. Moreover, when slashing delegators the tokens can only be burned and not used as slashing reward, this means that a malicious data service cannot profit directly from slashing delegators. In general, the more stake the service provider has on a provision the safer delegators will be. This protections in combination with per data service delegation make delegators safe from these types of attacks.
```

Comment:
> In general, the more stake the service provider has on a provision the safer delegators will be.

Is this true, considering the slashable amount is defined on a data service basis, as mentioned in the section below? Or is it generally true, considering that some rules are _enforced/(encouraged?)_ by the Data Service Framework?

Reply:
The comment above didn't take into account this part found below.

> As a general guideline however, the Data Service framework proposes the slashing amount should be proportional to the provisioned stake at the time a dispute is raised.

So I guess this is generally true if the Data Service follows the Frameworks' guidelines.